sysdeps/nacl/syscall.c

```diff
@@ -0,0 +1,34 @@
+#include <errno.h>
+#include <stdarg.h>
+#include <nacl_syscalls.h>
+#include <irt_syscalls.h>
+
+long int __syscall (long int __sysno, ...) {
+
+
+    if (__sysno < 0 || __sysno > NACL_MAX_SYSCALLS) {
+      __set_errno (ENOSYS);
+      return -1;
+    }
+
+    va_list args;
+    va_start(args, __sysno);
+    long arg1 = va_arg(args, long);
+    long arg2 = va_arg(args, long); 
+    long arg3 = va_arg(args, long); 
+    long arg4 = va_arg(args, long); 
+    long arg5 = va_arg(args, long); 
+    long arg6 = va_arg(args, long); 
+
+    int result = ((TYPE_nacl_syscall) NACL_SYSCALL_ADDR (__sysno)) (arg1, arg2, arg3, arg4, arg5, arg6);
```

Comment:
This is probably unspecified behavior... I don't believe you can get va_arg(s) beyond what's provided and while this will probably just be garbage, it's maybe worth thinking about.